lib/ex_doc/retriever.ex

```diff
@@ -357,9 +375,18 @@ defmodule ExDoc.Retriever do
     signature = signature(type_data.signature)
     annotations = if type_data.type == :opaque, do: ["opaque" | annotations], else: annotations
     doc_ast = doc_ast(content_type, doc, file: source.path)
+    metadata = Map.put(metadata, :__doc__, :type)
+    group = GroupMatcher.match_function(groups_for_docs, metadata)
+
+    id =
+      if name == nil do
+        "t:nil/#{arity}"
+      else
+        "t:#{name}/#{arity}"
```

Comment:
Perhaps worth creating a tiny function called `name_or_nil` that we can reuse across this module?

Reply:
done